packages/playwright-core/src/server/browserType.ts

```diff
@@ -112,6 +113,16 @@ export abstract class BrowserType extends SdkObject {
       args,
     });
     const [page] = context.pages();
+    // Chromium on macOS opens a new tab when clicking on the dock icon.
+    // See https://github.com/microsoft/playwright/issues/9434
+    if (this._name === 'chromium' && process.platform === 'darwin') {
```

Comment:
Can we move this whole `launchApp` method to a separate file? We can also merge things from `crApp` into it. This way this new code wouldn't seem out of place.